include/openssl/bio.h.in

```diff
@@ -520,7 +520,7 @@ int BIO_read_filename(BIO *b, const char *name);
 # define BIO_dup_state(b,ret)    BIO_ctrl(b,BIO_CTRL_DUP,0,(char *)(ret))
 
 # define BIO_reset(b)            (int)BIO_ctrl(b,BIO_CTRL_RESET,0,NULL)
-# define BIO_eof(b)              (int)BIO_ctrl(b,BIO_CTRL_EOF,0,NULL)
+# define BIO_eof(b)              (BIO_ctrl(b,BIO_CTRL_EOF,0,NULL) != 0)
```

Comment:
``` suggestion
# define BIO_eof(b)              (BIO_ctrl(b,BIO_CTRL_EOF,0,NULL) > 0)
```

At the very least, it avoids making EOF and errors equivalent

Reply:
No! All the current callers treat the negative return of BIO_eof() as if EOF was signaled and for a good reason. An EOF means to stop trying reading further. The current negative returns from BIO_eof() are unrecoverable, meaning no further call of BIO_eof() can succeed anyway. Returning 0 instead of -1 would signalize to the callers that it makes sense to try to read further and that would be wrong.

Reply:
It's quite obvious that changing things at this level is plain wrong.

-1

Reply:
Let's hand out duel guns :D